components/LessonPage/LessonEditor.jsx

```diff
@@ -1,252 +1,277 @@
-import {
-  clearCurrentCourse,
-  setCurrentCourse,
-} from '../../store/courses/reducer';
-import {
-  clearCurrentLesson,
-  setCurrentLesson,
-} from '../../store/lesson/reducer';
-import { useDispatch, useSelector } from 'react-redux';
-import { Button, Col, Container, Modal, Row } from 'react-bootstrap';
-import { updateLesson } from '../../store/lesson/thunks';
-import TextStepEditor from '../Steps/TextStep/TextStepEditor';
-import { getCurrentLesson } from '../../store/lesson/selectors';
-import { setSteps } from '../../store/lessonSteps/reducer';
-import { getSteps } from '../../store/lessonSteps/selectors';
-import VideoStepEditor from '../Steps/VideoStep/VideoStepEditor';
-import { useTranslation } from 'next-i18next';
-import Loader from '../Loader/Loader';
-import {
-  createTextAnswerLesson,
-  createTextLesson,
-  createVideoLesson,
-} from '../../store/lessonSteps/thunks';
-import OpenAnswerStepEditor from '../Steps/OpenAnswerStep/OpenAnswerStepEditor';
-import TestStepEditor from '../Steps/TestTaskStep/TestStepEditor';
-import { createTestStep } from '../../store/lessonSteps/test.thunk';
-import NewCourseNavbar from '../Navbars/NewCourseNavbar';
-import { Layout } from 'antd';
-const { Sider, Content } = Layout;
+import React, { useEffect, useState } from 'react'
+import { clearCurrentCourse, setCurrentCourse } from '../../store/courses/reducer'
+import { clearCurrentLesson, setCurrentLesson } from '../../store/lesson/reducer'
+import { useDispatch, useSelector } from 'react-redux'
+import { Button, Col, Container, Modal, Row } from 'react-bootstrap'
+import { updateLesson } from '../../store/lesson/thunks'
+import TextStepEditor from '../Steps/TextStep/TextStepEditor'
+import { getCurrentLesson } from '../../store/lesson/selectors'
+import { setSteps } from '../../store/lessonSteps/reducer'
+import { getSteps } from '../../store/lessonSteps/selectors'
+import VideoStepEditor from '../Steps/VideoStep/VideoStepEditor'
+import { useTranslation } from 'next-i18next'
+import Loader from '../Loader/Loader'
+import { createTextAnswerLesson, createTextLesson, createVideoLesson } from '../../store/lessonSteps/thunks'
+import OpenAnswerStepEditor from '../Steps/OpenAnswerStep/OpenAnswerStepEditor'
+import TestStepEditor from '../Steps/TestTaskStep/TestStepEditor'
+import CodeTaskStepEditor from '../Steps/CodeTaskStep/CodeTaskStepEditor'
+import { createTestStep } from '../../store/lessonSteps/test.thunk'
+import NewCourseNavbar from '../Navbars/NewCourseNavbar'
+import { Layout } from 'antd'
+
+const { Sider, Content } = Layout
 
 const stepTypes = {
   Text: TextStepEditor,
   Video: VideoStepEditor,
   TextWithAnswer: OpenAnswerStepEditor,
   Test: TestStepEditor,
-};
+  Code: CodeTaskStepEditor,
+}
 
 const stepIcons = {
   Text: 'fa-align-left',
   Video: 'fa-film',
   TextWithAnswer: 'fa-question',
   Test: 'fa-list-alt',
-};
+  Code: 'fa-code',
+}
+
+const LessonEditorModal = ({ show, setShow }) => {
```

Comment:
Maybe make a separate component for editor modal?

Reply:
okay, it has already done in another branch ;)